packages/apps/src/Menu/Item.tsx

```diff
@@ -18,10 +18,13 @@ const DUMMY_COUNTER = () => 0;
 
 function Item ({ className = '', isToplevel, route: { Modal, href, icon, name, text, useCounter = DUMMY_COUNTER } }: Props): React.ReactElement<Props> {
   const [isModalVisible, toggleModal] = useToggle();
+  const [colorState, setColorState] = useState('gray');
   const count = useCounter();
 
   return (
-    <li className={`${className}${count ? ' withCounter' : ''} ${isToplevel ? 'topLevel  highlight--color-contrast' : 'highlight--hover-color'}`}>
+    <li className={`${className}${count ? ' withCounter' : ''} ${isToplevel ? 'topLevel  highlight--color-contrast' : 'highlight--hover-color'}`}
+      onMouseEnter={() => setColorState('counter')}
```

Comment:
Please no inline functions. Where required, use `useCallback` - we don't want to recreate these on each render.